server/historian/packages/historian-base/src/routes/summaries.ts

```diff
@@ -187,12 +193,38 @@ export function create(
 					? request.query.initial
 					: request.query.initial === "true";
 
+			const isEphemeralFromRequest = request.get("IsEphemeralContainer");
+			// We treat these cases where we did not get the header as non-ephemeral containers
+			const isEphemeral: boolean =
+				typeof isEphemeralFromRequest === "undefined"
+					? false
+					: typeof isEphemeralFromRequest === "boolean"
+					? isEphemeralFromRequest
+					: isEphemeralFromRequest === "true";
+
+			// Only the initial post summary has a valid IsEphemeralContainer flag which we store in cache
```

Comment:
As discussed offline "The reason we need to do this is because this flag is sent only as a part of the initial summary upload..at this time this info is stored in the historian cache..for all the subsequent calls, we will need to get this info from the historian cache itself"

Reply:
I mentioned this in other comment but I think this is something important so apologies for the repetition here...

What if the cache overflows and the eviction policy removes the key from Redis? I think we should avoid using Redis as a database. The fact that the document is ephemeral is some sort of immutable property of that document. Storing in Redis is for volatile information that we could potentially lose. I am concerned of the potential side effects to those ephemeral documents if something like that happens.

Why can't we always make the caller provide the information whether a document is immutable or not? You mentioned "The reason we need to do this is because this flag is sent only as a part of the initial summary upload". Why is that the case? Can't we make it be present on every subsequent requests?
 - For the requests coming from service (alfred, scribe), we would be able to obtain that info from the database easily.
- For requests coming from client, would the driver changes being implemented be capable of adding that info in the request?

Reply:
Had the conversation with Henrique and Zach offline. We decided on caching with the db fallback the way it currently implemented to follow the precedent set by the storage name feature.